context/src/junit/bindings.rs

```diff
@@ -31,6 +32,193 @@ pub struct BindingsReport {
     pub test_suites: Vec<BindingsTestSuite>,
 }
 
+impl From<TestCaseRunStatus> for BindingsTestCaseStatusStatus {
+    fn from(value: TestCaseRunStatus) -> Self {
+        match value {
+            TestCaseRunStatus::Success => BindingsTestCaseStatusStatus::Success,
+            TestCaseRunStatus::Failure => BindingsTestCaseStatusStatus::NonSuccess,
+            TestCaseRunStatus::Skipped => BindingsTestCaseStatusStatus::Skipped,
+            TestCaseRunStatus::Unspecified => BindingsTestCaseStatusStatus::Unspecified,
+        }
+    }
+}
+
+impl From<TestResult> for BindingsReport {
+    fn from(
+        TestResult {
+            test_case_runs,
+            uploader_metadata,
+        }: TestResult,
+    ) -> Self {
+        let test_cases: Vec<BindingsTestCase> = test_case_runs
+            .into_iter()
+            .map(|testcase| BindingsTestCase::from(testcase))
```

Comment:
totally optional: this can be simplified to
```suggestion
            .map(BindingsTestCase::from)
```